src/asm/opt.c

```diff
@@ -17,7 +17,7 @@ struct OptStackEntry {
 	bool haltnop;
 	bool optimizeLoads;
 	bool warningsAreErrors;
-	enum WarningState warningStates[NB_WARNINGS];
+	enum WarningState warningStates[sizeof(warningStates)];
```

Comment:
I instead went with a comment to explain; the advantage of doing it like this is that this will auto-update, reducing some coupling.

Reply:
`enum WarningState warningStates` is declared as `[NB_PLAIN_WARNINGS + NB_PARAM_WARNINGS]` (although as I commented above, this could be `[NB_WARNINGS_PARAM]`, or rename to `[NB_PLAIN_AND_PARAM_WARNINGS]`). So using the same constant here would still keep them in sync.